elastic/beats
# [Enhancement] Add RotateOnStartup feature flag for file output

## What does this PR do?
File output can now be configured to disable file rotation on startup. This PR introduces the configuration option `rotate_on_startup` and configures the internally used FileRotator (rotator.go) accordingly.

## Why is it important?
To avoid file rotation on every startup.

## Checklist


~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist
- [ ] check whether the name of the added configuration option 'rotate_on_startup' conforms to guideline
- [ ] test that default behaviour has not changed, i.e. ouput file is rotated on every startup

## How to test this PR locally
Configure beat to file output. If configuration option `rotate_on_startup` is set to `false`, file may no longer be rotated on startup.

## Related issues
- Closes elastic/beats#19150
